src/targets/gpu/lowering.cpp

```diff
@@ -505,6 +508,60 @@ struct miopen_apply
         });
     }
 
+    void add_scatternd()
+    {
+        apply_map.emplace("scatternd_none", [=](instruction_ref ins) {
+            auto s      = ins->get_shape();
+            auto op_val = ins->get_operator().to_value();
+            auto output = insert_allocation(ins, s);
+            auto args   = ins->inputs();
+            args.push_back(output);
+
+            auto io_shapes = to_shapes(args);
+            check_shapes{io_shapes, ins->get_operator()}.standard();
```

Comment:
The shapes should already be standard here, so this check should be unnecessary.

Reply:
How the shape has been standard here?

Reply:
Because auto_contiguous ensures everything is standard shape before lowering.

Reply:
But the eliminate_contiguous can remove the contiguous later, and then the computation is wrong.

Reply:
It wont because the code_object_op will throw an error if the shapes change(see [here](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/targets/gpu/code_object_op.cpp#L21)).